### PR TITLE
Fix `target` leakage in `MeanSegmentEncoderTransform`

### LINKED ISSUE #400: [BUG] Target leakage in `MeanSegmentEncoderTransform`
### 🐛 Bug Report

Now mean is counting including current timestamp. It should not do this.

### Expected behavior

Mean is counting not including current timestamp

### Code
```
from etna.datasets import TSDataset
from etna.datasets import generate_ar_df
from etna.transforms import MeanSegmentEncoderTransform

df = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=5)
df["target"] = range(5)
ts = TSDataset(df, freq="D")
tr = MeanSegmentEncoderTransform()
ts.fit_transform([tr])
ts.df.loc[:, pd.IndexSlice[:, "segment_mean"]].values
```
### Output
```
array([[0. ],
       [0.5],
       [1. ],
       [1.5],
       [2. ]])
```
### Expected output
```
array([[0. ],
       [0. ],
       [0.5],
       [1. ],
       [1.5]])
```

### Environment

_No response_

### Additional context

_No response_

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by egoriyaa**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues
closes #400

### CODE REVIEW

File: tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py [hunk 1]
```diff
@@ -10,44 +11,31 @@
...
 
 
-def test_mean_segment_encoder_transform(simple_ts, transformed_simple_df):
-    encoder = MeanSegmentEncoderTransform()
-    transformed_df = encoder.fit_transform(simple_ts).to_pandas()
-    transformed_simple_df.index.freq = "D"
-    pd.testing.assert_frame_equal(transformed_simple_df, transformed_df)
-
-
-def test_subset_segments(simple_ts):
```
Comment (egoriyaa) [inline tests/test_transforms/test_encoders/test_mean_segment_encoder_transform.py:27]: We have such type of tests in inference tests

File: tests/test_transforms/test_encoders/test_segment_encoder_transform.py [hunk 2]
```diff
@@ -2,54 +2,36 @@
...
-    for segment in simple_ts.segments:
+    for segment in mean_segment_encoder_ts.segments:
         column = transformed_df.loc[:, pd.IndexSlice[segment, "segment_code"]]
         assert column.dtype == "category", "Column type is not category"
         assert np.all(column == column.iloc[0]), "Values are not the same for the whole column"
         codes.add(column.iloc[0])
     assert codes == {0, 1}, "Codes are not 0 and 1"
 
 
-def test_subset_segments(simple_ts):
```
Comment (egoriyaa) [inline tests/test_transforms/test_encoders/test_segment_encoder_transform.py:27]: The same as for `MeanSegmentEncoder`

File: etna/transforms/encoders/mean_segment_encoder.py [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
     idx = pd.IndexSlice
+    segment_column = "segment_column"
+    out_column = "segment_mean"
 
     def __init__(self):
         super().__init__(required_features=["target"])
-        self.mean_encoder = MeanTransform(in_column="target", window=-1, out_column="segment_mean")
-        self.global_means: Optional[Dict[str, float]] = None
+        self.mean_encoder = MeanEncoderTransform(
+            in_column=self.segment_column, mode="per-segment", out_column=self.out_column, smoothing=0
```
Comment (d-a-bunin) [inline etna/transforms/encoders/mean_segment_encoder.py:20]: Don't we want to add `smoothing` parameter?

File: etna/transforms/encoders/mean_segment_encoder.py [hunk 4]
```diff
@@ -1,24 +1,35 @@
...
 
     idx = pd.IndexSlice
+    segment_column = "segment_column"
+    out_column = "segment_mean"
 
     def __init__(self):
         super().__init__(required_features=["target"])
-        self.mean_encoder = MeanTransform(in_column="target", window=-1, out_column="segment_mean")
-        self.global_means: Optional[Dict[str, float]] = None
+        self.mean_encoder = MeanEncoderTransform(
```
Comment (d-a-bunin) [inline etna/transforms/encoders/mean_segment_encoder.py:19]: Why is it public?

File: etna/transforms/encoders/mean_segment_encoder.py [hunk 5]
```diff
@@ -1,24 +1,35 @@
...
 from etna.transforms import IrreversibleTransform
-from etna.transforms.math.statistics import MeanTransform
+from etna.transforms.encoders.mean_encoder import MeanEncoderTransform
 
 
 class MeanSegmentEncoderTransform(IrreversibleTransform):
     """Makes expanding mean target encoding of the segment. Creates column 'segment_mean'."""
 
     idx = pd.IndexSlice
+    segment_column = "segment_column"
```
Comment (d-a-bunin) [inline etna/transforms/encoders/mean_segment_encoder.py:14]: Why is it public?

File: tests/test_transforms/test_encoders/conftest.py [hunk 6]
```diff
@@ -1,38 +1,34 @@
...
-    df_1["segment"] = "Moscow"
-    df_1["target"] = [1.0, 2.0, 3.0, 4.0, 5.0, np.NAN, np.NAN]
-    df_1["exog"] = [6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0]
-    df_2["segment"] = "Omsk"
-    df_2["target"] = [10.0, 20.0, 30.0, 40.0, 50.0, np.NAN, np.NAN]
-    df_2["exog"] = [60.0, 70.0, 80.0, 90.0, 100.0, 110.0, 120.0]
-    classic_df = pd.concat([df_1, df_2], ignore_index=True)
-    df = TSDataset.to_dataset(classic_df)
-    ts = TSDataset(df, freq="D")
+def mean_segment_encoder_ts() -> TSDataset:
```
Comment (d-a-bunin) [inline tests/test_transforms/test_encoders/conftest.py:9]: Why are we putting this fixture here?

File: etna/transforms/encoders/mean_segment_encoder.py (hunk 3 above)
Comment (egoriyaa) [reply etna/transforms/encoders/mean_segment_encoder.py:20]: Where?

File: tests/test_transforms/test_encoders/conftest.py (hunk 6 above)
Comment (egoriyaa) [reply tests/test_transforms/test_encoders/conftest.py:9]: I use this fixture both in tests with `MeanEncoderTransofrm` and `MeanSegmentEncoderTransofrm`

File: etna/transforms/encoders/mean_segment_encoder.py (hunk 3 above)
Comment (d-a-bunin) [reply etna/transforms/encoders/mean_segment_encoder.py]: I'm suggesting that we could add `smoothing` parameter into `MeanSegmentEncoderTransform`.

Comment (egoriyaa) [reply etna/transforms/encoders/mean_segment_encoder.py]: See my previous comment

**Review by d-a-bunin — APPROVED** [review]

**Pull request merged**